serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java

```diff
@@ -187,3 +228,106 @@ public E deserializeNullable(@NonNull Decoder decoder, @NonNull DecoderContext c
+    }
+
+    @NonNull
+    private E transform(@NonNull Decoder decoder, Object value) throws IOException {
+        for (EnumBeanIntrospection.EnumConstant<? super E> enumConstant : enumIntrospection.getConstants()) {
+            E enumValue = (E) enumConstant.getValue();
+            Object extractedValue = valueExtractor.apply(enumValue);
+            if (Objects.equals(extractedValue, value)) {
+                return enumValue;
+            }
+        }
+
+        var allowedValues = enumIntrospection.getConstants().stream()
+            .map(EnumBeanIntrospection.EnumConstant::getValue)
+            .map(enumValue -> valueExtractor.apply((E) enumValue))
+            .map(Object::toString)
+            .collect(Collectors.joining(", "));
+
+        throw decoder.createDeserializationException("Expected one of [%s] but was '%s'".formatted(allowedValues, value), value);
+    }
+
+    @Override
+    public E deserialize(@NonNull Decoder decoder, @NonNull DecoderContext context, @NonNull Argument<? super E> type) throws IOException {
+        return transform(decoder, valueDeserializer.deserialize(decoder, context, valueType));
+    }
+
+    @Override
+    public E deserializeNullable(@NonNull Decoder decoder, @NonNull DecoderContext context, @NonNull Argument<? super E> type) throws IOException {
+        Object v = valueDeserializer.deserializeNullable(decoder, context, valueType);
+        if (!allowNull && v == null) {
+            return null;
+        }
+        return transform(decoder, v);
+    }
+}
+
+final class EnumPropertySerializer<E extends Enum<E>> implements Serializer<E> {
+
+    private final EnumBeanIntrospection<? extends E> enumIntrospection;
+
+    EnumPropertySerializer(EnumBeanIntrospection<? extends E> enumIntrospection) {
+        this.enumIntrospection = enumIntrospection;
+    }
+
+    @Override
+    public void serialize(@NonNull Encoder encoder, @NonNull EncoderContext context, @NonNull Argument<? extends E> type, E value) throws IOException {
+        for (EnumBeanIntrospection.EnumConstant<? extends E> enumConstant : enumIntrospection.getConstants()) {
+            if (enumConstant.getValue() == value) {
+                encoder.encodeString(enumConstant.stringValue(SerdeConfig.class, SerdeConfig.PROPERTY).orElse(value.name()));
+            }
+        }
+    }
+}
+
+final class EnumPropertyDeserializer<E extends Enum<E>> implements Deserializer<E> {
+
+    private final EnumBeanIntrospection<? super E> enumIntrospection;
+
+    EnumPropertyDeserializer(EnumBeanIntrospection<? super E> enumIntrospection) {
+        this.enumIntrospection = enumIntrospection;
+    }
+
+
+    @Override
+    public E deserialize(@NonNull Decoder decoder, @NonNull DecoderContext context, @NonNull Argument<? super E> type) throws IOException {
+        var value = decoder.decodeString();
+
+        for (EnumBeanIntrospection.EnumConstant<? super E> enumConstant : enumIntrospection.getConstants()) {
```

Comment:
Same here, please try to precalculate

Reply:
Added pre-calculation in 9d4e0890d6e775e04edaa008ebab5a90e8316a40